uarch/uarch-runtime.cpp

```diff
@@ -70,13 +70,12 @@ extern "C" void _putchar(char c) {
     *p = c;
 }
 
-extern "C" [[noreturn]] void abort(void) {
+extern "C" NO_RETURN void abort(void) {
     volatile char *p = reinterpret_cast<char *>(cartesi::uarch_mmio_address::abort);
     *p = uarch_mmio_abort_value;
     // execution will never reach this point
-    // infinite loop added to silence the compiler
-    for (;;) {
-    }
+    asm("1: j 1b"); // infinite loop
```

Comment:
When coding `asm` statements, you should use `asm volatile`, otherwise the compiler is free to change or optimize our assembly code.

Reply:
I don't think we need the infinite look, right? What is the side effect we'd like to have with `abort`? Do we want to pass out the information that the microarch emulator halted with an error condition?

Reply:
I also feel that `__builtin_trap()` would make more sense here rather than an infinite loop, but again both are pointless, because the interpreter will halt before reaching it.